windows/ReactNativeWebView/ReactWebViewManager.cpp

```diff
@@ -0,0 +1,173 @@
+#include "pch.h"
+#include "ReactWebViewManager.h"
+#include "NativeModules.h"
+
+namespace winrt {
+    using namespace Microsoft::ReactNative;
+    using namespace Windows::Foundation;
+    using namespace Windows::Foundation::Collections;
+    using namespace Windows::UI::Xaml;
+    using namespace Windows::UI::Xaml::Controls;
+}
+
+namespace winrt::ReactNativeWebView::implementation {
+
+    ReactWebViewManager::ReactWebViewManager() {}
+
+    // IViewManager
+    winrt::hstring ReactWebViewManager::Name() noexcept {
+        return L"RCTWebView";
+    }
+
+    winrt::FrameworkElement ReactWebViewManager::CreateView() noexcept {
+        auto webView = winrt::WebView();
+        m_webView = winrt::make_weak(webView);
+        RegisterEvents();
+        return webView;
+    }
+
+    void ReactWebViewManager::RegisterEvents() {
+        if (auto webView = m_webView.get()) {
+            m_navigationCompletedRevoker = webView.NavigationCompleted(
+                winrt::auto_revoke, [=](auto const& sender, auto const& args) {
+                    OnNavigationCompleted(sender, args);

```

Comment:
I haven't read enough yet to know that this is actually a bug yet, but capturing `this` in a lambda that is going to be called asynchronously from some other class is often incorrect. If you're 100% sure that `this` will always outlive `webView` then this is OK.

If the lambda outlives `this`, that's a crash right here.

One way to avoid this bug is to use `shared_from_this()`, and the lambda can keep a weak_ptr to the ReactWebViewManager and try to `lock()` it before calling `OnNavigationCompleted`.

Reply:
It looks like the lifetime of webView is not necessarily properly aligned with the lifetime of ReactWebViewManager. This class only owns a weak_ref to the WebView. I'm not super familiar with the IViewManager interface. Does it guarantee that the ViewManager will outlive the view returned by `CreateView`?

Reply:
Good catch! I think ViewManager should outlive WebView, I've changed the weak ref to strong ref.